<?xml version="1.0"?>
<launch>
    <!-- Gazebo arguments -->
    <arg name="gui" default="true"/>
    <arg name="paused" default="true"/>

    <!-- RViz -->
    <arg name="use_rviz" default="false"/>

    <!-- ROS Bag record options -->
    <arg name="record" default="false"/>
    <arg name="duration" default="30s"/>
    <arg name="bag_dir" default="/tmp/"/>
    <arg name="bag_name_prefix" default="collect_images"/>

    <!-- Periodically reset Gazebo model pose -->
    <arg name="reset_pose" default="true"/>
    <arg name="reset_period" default="0.25"/>
    <arg name="num_scenes_each_truth" default="100"/>
    <arg name="truth_file"
         default="$(find gem_scenario_runner)/configs/example-5.truths.yaml"/>

    <!-- Constants -->
    <arg name="world_name"
         default="$(find gem_scenario_runner)/worlds/three_100m_straight_roads.world"/>
    <arg name="robot_name" value="gem"/>

    <include file="$(find gem_gazebo)/launch/gem_gazebo_rviz.launch">
        <arg name="world_name" value="$(arg world_name)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_rviz" value="$(arg use_rviz)"/>
    </include>

    <group ns="$(arg robot_name)" if="$(arg reset_pose)">
        <node name="set_model_pose" pkg="gem_scenario_runner" type="set_scenes.py"
              output="screen" required="true">
            <param name="gazebo_world_name" value="$(arg world_name)"/>
            <param name="gazebo_model_name" value="$(arg robot_name)"/>
            <param name="reset_period" type="double" value="$(arg reset_period)"/>
            <param name="num_scenes_each_truth" type="int" value="$(arg num_scenes_each_truth)"/>
            <rosparam command="load" file="$(arg truth_file)"/>
        </node>
    </group>

    <group ns="$(arg robot_name)" if="$(arg record)">
        <!-- ROS Bag record options -->
        <arg name="record_topics" default="base_footprint/odom front_single_camera/image_raw/compressed"/>
        <arg name="buffsize" value="256"/>
        <node name="$(anon record)" pkg="rosbag" type="record" output="screen" required="true"
              args="--buffsize $(arg buffsize) --duration $(arg duration) --tcpnodelay -o $(arg bag_dir)/$(arg bag_name_prefix) $(arg record_topics)"/>
    </group>
</launch>
